test/test_sparqlstore.py

```diff
@@ -74,19 +77,21 @@ def test_counting_graph_and_store_queries(self):
             """
         g = Graph("SPARQLStore")
         g.open(self.path)
-        c = 0
-        for r in g.query(q):
-            c += 1
+        count = 0
+        result = helper.query_with_retry(g, query)
+        for _ in result:
+            count += 1
```

Comment:
If by renaming you meant to leave code in better shape than you found it, why not simply:
```py
result = ...
count = len(result)
```

Reply:
actually `len` only works if the result is already retrieved. The only reason it works in this context is this line:

https://github.com/RDFLib/rdflib/pull/1203/files#diff-4dd682d5c858fe7d3f39400e3da7a0ea9e77232bab93c6d51e5cb523844665b2R42

As that is somewhat counter intuitive, I will fix `len` in a separate MR. Afterwards I am fine with removing those for loops (there are more of those in the tests).

Reply:
Didn't check, just though it should work based on these `query.Result` lines:
https://github.com/RDFLib/rdflib/blob/56dc4207ce6e7b11ed7b45fb4fd4020ba548e718/rdflib/query.py#L245-L251

Alternatively, even something like `count = len(list(result))` would be better.